src/Redux/books/bookSlice.js

```diff
@@ -1,49 +1,100 @@
-// ./src/redux/books/booksSlice.js
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-import { createSlice } from '@reduxjs/toolkit';
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/2uLpDoKDZ8I23S6ajxuI';
+// 2uLpDoKDZ8I23S6ajxuI
 
-const initialState = [
-  {
-    item_id: 'item1',
-    title: 'Lone Women',
-    author: 'Ema Cline',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item2',
-    title: 'Romantic Comedy',
-    author: 'Mathew Desmond',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item3',
-    title: 'The Covenant of a Woman',
-    author: 'Abraham Verghese',
-    category: 'Nonfiction',
+const initialState = {
+  books: {},
```

Comment:
- [ ] You've made significant progress with your book add and remove functionality, which is working smoothly. However, in the previous milestone, books were stored as an array of objects, with each book represented as an individual object. I understand that the API provides data in a different format. So rather than altering the way books are stored in the Redux store, I recommend adding functionality to transform the API output into an array of objects and then storing it in the Redux store, as you did in the previous milestone as per [project requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api.md#:~:text=Note%3A%20Everything%20should%20work%20the%20same%20as%20before).

Currently, your `books` are stored like this:

```javascript
books: {
  '128b6a4d-9dd7-4d6b-ab65-fd29fc7448de': [
    {
      author: 'Nameless Trichor',
      title: 'Whats my Name',
      category: 'Philosophy'
    }
  ],
  'eaf57c7b-083d-4728-a5e1-d53b6759fa30': [
    {
      author: 'new author',
      title: 'new book',
      category: ''
    }
  ]
}
```

To achieve the desired format, you'll need to add functionality to transform `books` into an array like this:

```javascript
books= [
  {
    item_id: '128b6a4d-9dd7-4d6b-ab65-fd29fc7448de',
    title: 'Whats my Name',
    author: 'Nameless Trichor',
    category: 'Philosophy',
  },
  {
    item_id: 'eaf57c7b-083d-4728-a5e1-d53b6759fa30',
    title: 'new book',
    author: 'new author',
    category: '',
  },
];
```

This approach will maintain consistency with your previous milestone and ensure compatibility with your Redux store structure. Keep up the great work, and thank you for your attention to maintaining a structured and organized data format! 🌟📚